File: Haskell/HelloWorld/src/Kex2/Kex2.hs
module Kex2.Kex2 where

import Prelude
import Data.Text (unpack)
import Shared.Player (Player(..))
import Shared.Constants (playerInitialHp, playerInitialDefense)
import Kex2.Kex2Battle (gameLoop)

run :: IO ()
run = do
  putStrLn "冒険が今始まる！"
  let player = Player{ playerHp = playerInitialHp, playerDefense = playerInitialDefense }
  gameResult <- gameLoop player 0 1
  putStrLn $ unpack gameResult

File: FSharp/HelloWorld/Lib/Kex2/Kex2.fs
module HelloWorld.Lib.Kex2.Kex2

open System
open HelloWorld.Lib.Shared

let runKex2 () =
    printfn "冒険が今始まる！"
    let player = { Player.Hp = Constants.PlayerInitialHp; Player.Defense = Constants.PlayerInitialDefense }
    Console.WriteLine (Kex2Battle.GameLoop player)